test/e2e/suite/peered_vpc.go

```diff
@@ -278,6 +278,16 @@ func (t *PeeredVPCTest) NewTestNode(ctx context.Context, instanceName, nodeName,
 	}
 }
 
+func (t *PeeredVPCTest) NewNodeMonitoringAgentTest() *addon.NodeMonitoringAgentTest {
```

Comment:
instead of polluting `PeeredVPCTest` with knowledge about add-on tests, what do you think about wrapping it (like `AddOnEC2Test`) and add these helper methods in there?

Reply:
i created `test/e2e/suite/addon_ec2.go` and moved it there